File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FAlertDialog.java
package io.github.gofaith.faithdroid.FViews;

import android.content.DialogInterface;
import android.support.v7.app.AlertDialog;

import org.json.JSONArray;
import org.json.JSONTokener;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FAlertDialog extends FView implements AttrSettable, AttrGettable {
    public AlertDialog.Builder v;
    public DialogInterface di;
    public FAlertDialog(UIController controller) {
        parentController=controller;
        v = new AlertDialog.Builder(parentController.activity);
    }
    @Override
    public String getAttr(String attr) {
        return null;
    }

    @Override
    public void setAttr(String attr, String value) {
        if (value==null)
            return;
        switch (attr) {
            case "Title":
                v.setTitle(value);
                break;
            case "View":
                if (parentController.viewmap.containsKey(value)) {
                    v.setView(parentController.viewmap.get(value).view);
                }
                break;
            case "PositiveButton":
                try {
                    JSONArray array = (JSONArray) (new JSONTokener(value).nextValue());
                    final String fnId = array.getString(1);
                    v.setPositiveButton(array.getString(0), new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            Faithdroid.triggerEventHandler(fnId, "");
                        }
                    });
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "NegativeButton":
                try {
                    JSONArray array = (JSONArray) (new JSONTokener(value).nextValue());
                    final String fnId = array.getString(1);
                    v.setNegativeButton(array.getString(0), new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            Faithdroid.triggerEventHandler(fnId, "");
                        }
                    });
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "Show":
                di = v.show();
                break;
            case "Dismiss":
                if (di!=null) {
                    di.dismiss();
                }
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FAlertDialog.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.content.DialogInterface
import androidx.appcompat.app.AlertDialog
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.UIController
import org.json.JSONArray
import org.json.JSONTokener

class FAlertDialog(controller: UIController) : FView(), AttrSettable, AttrGettable {
    var v: AlertDialog.Builder
    var di: DialogInterface? = null

    init {
        parentController = controller
        v = AlertDialog.Builder(parentController!!.activity)
    }

   override fun getAttr(attr: String): String {
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (value == null)
            return
        when (attr) {
            "Title" -> v.setTitle(value)
            "View" -> if (parentController!!.viewmap.containsKey(value)) {
                v.setView(parentController!!.viewmap.get(value)!!.view)
            }
            "PositiveButton" -> try {
                val array = JSONTokener(value).nextValue() as JSONArray
                val fnId = array.getString(1)
                v.setPositiveButton(
                    array.getString(0),
                    DialogInterface.OnClickListener { dialog, which -> Faithdroid.triggerEventHandler(fnId, "") })
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "NegativeButton" -> try {
                val array = JSONTokener(value).nextValue() as JSONArray
                val fnId = array.getString(1)
                v.setNegativeButton(
                    array.getString(0),
                    DialogInterface.OnClickListener { dialog, which -> Faithdroid.triggerEventHandler(fnId, "") })
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "Show" -> di = v.show()
            "Dismiss" -> if (di != null) {
                di!!.dismiss()
            }
        }
    }
}
